src/content/docs/paper/dev/api/dialogs.mdx

```diff
@@ -443,3 +470,71 @@ DialogAction.customClick(
         .build()
 )
 ```
+
+### Using NBT payloads
+
+Instead of using callbacks, you can use NBT payloads and event handlers.
+
+If we take a look at the dialog creation code again, the confirmation button is created like this:
+
+```java showLineNumbers
+    ActionButton.create(
+        Component.text("Confirm", TextColor.color(0xAEFFC1)),
+        Component.text("Click to confirm your input."),
+        100,
+        DialogAction.customClick(Key.key("papermc:user_input/confirm"), null)
+    ),
```

Comment:
would probably dedent this
```suggestion
ActionButton.create(
    Component.text("Confirm", TextColor.color(0xAEFFC1)),
    Component.text("Click to confirm your input."),
    100,
    DialogAction.customClick(Key.key("papermc:user_input/confirm"), null)
)
```

Reply:
Resolved in commit https://github.com/PaperMC/docs/pull/702/commits/4135b422dab0aa65b9a916464c3eb6888cf80e8f